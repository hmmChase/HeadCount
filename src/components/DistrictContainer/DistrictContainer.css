.cardContainer {
  display: -ms-flexbox;
  display: flex;
  -ms-flex-pack: justify;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  justify-content: space-around;
  width: 100%;
  padding-top: 2vh;
}
