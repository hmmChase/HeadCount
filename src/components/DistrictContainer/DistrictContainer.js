import React from 'react';
import Card from '../Card/Card';
import './DistrictContainer.css';

const DistrictContainer = ({ foundDistricts }) => {
  const districtCards = foundDistricts.map((district, index) => {
    return <Card key={`Card ${index}`} {...district} />;
  });

  return <section className='cardContainer'>{districtCards}</section>;
};

export default DistrictContainer;
