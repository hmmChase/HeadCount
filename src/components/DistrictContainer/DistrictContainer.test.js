import React from 'react';
import DistrictContainer from './DistrictContainer';
import { shallow, mount } from 'enzyme';
const mockProps = {
  comparedDistricts: [1, 2],
  foundDistricts: [],
  changeSelectedDistricts: jest.fn()
};

describe('DistrictsContainer', () => {
  let container;
  beforeEach(() => {
    container = shallow(<DistrictContainer {...mockProps} />);
  });

  it('matches snapshot', () => {
    expect(container).toMatchSnapshot();
  });

  it('assigns a clicked class to the appropriate card', () => {
    container = mount(<DistrictContainer 
      {...mockProps} 
      comparedDistricts={[{ average: 0.407, location: "ACADEMY 20", stats: {}}]} foundDistricts={[{ average: 0.407, location: "ACADEMY 20", stats: {}}]} />);

    const card = container.find('article');
    expect(card.hasClass('clicked')).toBe(true);
  });
});
