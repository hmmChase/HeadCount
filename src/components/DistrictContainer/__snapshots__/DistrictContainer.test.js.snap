// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DistrictsContainer matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DistrictContainer
    changeSelectedDistricts={[Function]}
    comparedDistricts={
        Array [
            1,
            2,
          ]
    }
    foundDistricts={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [],
      "className": "cardContainer",
    },
    "ref": null,
    "rendered": Array [],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [],
        "className": "cardContainer",
      },
      "ref": null,
      "rendered": Array [],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
